Template fixes

dim-37 opened the pull request.

Commit:
Added a workflow that builds a NuGet package and pushes it into MyGet

Commit:
Renamed Samples.sln

xperiandri committed:
Implemented validation errors as observable collections of objects

xperiandri committed:
Implemented updateObservableCollection instead of elmStyleMerge

Commit:
Added template implementation for visual studio

dim-37 committed:
Fixed template

xperiandri committed:
Fixed linker configuration for WASM

dim-37 committed:
Added Solution.Build to csproj files

xperiandri commented on src/Templates/SolutionTemplate/.template.config/template.json:
```diff
@@ -4,12 +4,12 @@
         "type": "project"
     },
-    "sourceName": "SolutionTemplate",
+    "sourceName": "App13",
```
Это надо откатить

xperiandri commented on src/Templates/SolutionTemplate/.vsconfig:
```diff
@@ -0,0 +1,112 @@
+    "Microsoft.Net.Component.4.7.1.SDK",
+    "Microsoft.Net.Component.4.7.2.SDK"
+  ]
+}
```
Это я почищу

xperiandri replied:
Мои изменения зайлей. То что я в чате присылал

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Droid/Properties/AssemblyInfo.cs:
```diff
@@ -1,14 +1,16 @@
 // set of attributes. Change these attribute values to modify the information
 // associated with an assembly.
-[assembly: AssemblyTitle(AssemblyInfo.Product + " UWP app")]
-[assembly: AssemblyDescription("UWP приложение"+ AssemblyInfo.Description)]
```
Надо откатить до этого. Только оставить
```
[assembly: AssemblyTitle(AssemblyInfo.Product + " UWP app")]
/// TODO: Modify description per platform
[assembly: AssemblyDescription(AssemblyInfo.Description)]
```

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Droid/SolutionTemplate.Droid.csproj:
```diff
@@ -1,27 +1,23 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\Solution.Build.props" />
+	<Import Project="..\..\Solution.Build.props" />
```
По всем файлам проектов пройтись `Ctrl+E,D` и untabify

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Droid/SolutionTemplate.Droid.csproj:
```diff
@@ -1,27 +1,23 @@
     <OutputType>Library</OutputType>
     <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>$(AssemblyBaseName)</RootNamespace>
-    <AssemblyName>$(AssemblyBaseName).Droid</AssemblyName>
```
Вернуть

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Droid/SolutionTemplate.Droid.csproj:
```diff
@@ -105,16 +106,20 @@
     <None Include="Properties\AndroidManifest.xml" />
   </ItemGroup>
+  <ItemGroup>
+    <AndroidResource Include="Resources\values\Styles.xml" />
```
Я все ресурсы консолидировал рядом по алфавиту. Сделай так же

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Shared/App.xaml.cs:
```diff
@@ -3,43 +3,34 @@
     /// </summary>
+#pragma warning disable CA1724
     public sealed partial class App : Application
+#pragma warning restore CA1724
```
По идее все эти предупреждения должны быть выключены в `.editorconfig` по файловому шаблону поиска

xperiandri replied:
Я имею ввиду те, которые не нужны в конкретном файле. Например использовать лямбды если метод в 1 строку. Для элементов управления и страниц это выключено

dim-37 replied:
без этого WASM проект не строиться

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Shared/App.xaml.cs:
```diff
@@ -171,15 +174,17 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
 #if __WASM__
+#pragma warning disable DF0000 // Marks undisposed anonymous objects from object creations.
                 builder.AddProvider(new global::Uno.Extensions.Logging.WebAssembly.WebAssemblyConsoleLoggerProvider());
+#pragma warning restore DF0000 // Marks undisposed anonymous objects from object creations.
```
Там в анализаторе орфографическая ошибка. Поменяй везде `undisposed` на `indisposed`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Shared/Controls/ValidationControl.cs:
```diff
@@ -18,7 +18,9 @@
 {
+#pragma warning disable CA1010 // Generic interface should also be implemented
     public sealed partial class ValidationControl : ContentControl
+#pragma warning restore CA1010 // Generic interface should also be implemented
```
Вот такие штуки должны быть выключены на control'ах через `.editorconfig`

dim-37 replied:
тоже самое если убрать будет ломаться WASM проект

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Shared/Controls/ValidationControl.cs:
```diff
@@ -109,7 +113,13 @@ public Style ErrorContentStyle
         /// <summary>
         /// Provides derived classes an opportunity to handle changes to the ErrorContentStyle property.
         /// </summary>
+#pragma warning disable CS0628 // New protected member declared in sealed type
```
Он не должен быть `protected`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Gtk/SolutionTemplate.Skia.Gtk.csproj:
```diff
@@ -1,14 +1,11 @@
   <PropertyGroup>
     <OutputType Condition="'$(Configuration)'=='Release'">WinExe</OutputType>
     <OutputType Condition="'$(Configuration)'=='Debug'">Exe</OutputType>
-    <LangVersion>9.0</LangVersion>
```
Вернуть 9 C# во все C# проекты

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Gtk/SolutionTemplate.Skia.Gtk.csproj:
```diff
@@ -1,14 +1,11 @@
   </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)'=='Release'">
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-  </PropertyGroup>
```
Ты это в `build.props` вынес?

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Gtk/SolutionTemplate.Skia.Gtk.csproj:
```diff
@@ -18,31 +15,36 @@
+    <PackageReference Include="Uno.Microsoft.Toolkit.Uwp" Version="7.0.*" />
+    <PackageReference Include="Uno.Microsoft.Toolkit.Uwp.UI" Version="7.0.*" />
+    <PackageReference Include="Uno.Microsoft.Toolkit.Uwp.UI.Controls" Version="7.0.*" />
+
```
Без пустых строк и по алфавиту

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Wpf/SolutionTemplate.Skia.Wpf.csproj:
```diff
@@ -1,44 +1,45 @@
-    <LangVersion>9.0</LangVersion>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
-    <UseWPF>true</UseWPF>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Давай на .NET 5 переедем

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Wpf/SolutionTemplate.Skia.Wpf.csproj:
```diff
@@ -1,44 +1,45 @@
-    <PackageReference Include="Uno.UI.RemoteControl" Version="3.7.*" Condition="'$(Configuration)'=='Debug'" />
   </ItemGroup>
   <ItemGroup>
-    <Content Include="Assets\Fonts\uno-fluentui-assets.ttf" />
```
А шрифты не потерялись? Или они не нужны уже?

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.UWP/Properties/AssemblyInfo.cs:
```diff
@@ -1,15 +1,14 @@
 // associated with an assembly.
-[assembly: AssemblyTitle(AssemblyInfo.Product + " Android app")]
-[assembly: AssemblyDescription("Android приложение" + AssemblyInfo.Description)]
+[assembly: AssemblyTitle("SolutionTemplate.UWP")]
```
Аналогично с исправлением Android на UWP

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.UWP/Properties/Default.rd.xml:
```diff
@@ -21,11 +21,11 @@
-    
     <!-- Add your application specific runtime directives here. -->
 
 
```
Без этих пустых строк

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.UWP/SolutionTemplate.UWP.csproj:
```diff
@@ -1,49 +1,52 @@
       <!--
       If, in the same solution, you are referencing a project that uses https://github.com/onovotny/MSBuildSdkExtras,
-      you need to make sure that the version provided here matches https://github.com/onovotny/MSBuildSdkExtras/blob/master/Source/MSBuild.Sdk.Extras/DefaultItems/ImplicitPackages.targets#L11.
+      you need to make sure that the version provided here matches https://github.com/novotnyllc/MSBuildSdkExtras/blob/main/Source/MSBuild.Sdk.Extras/DefaultItems/ImplicitPackages.targets#L11.
```
О, так они переехали в другой репозиторий и 11 таки поддерживается. Надо у нас тогда в примерах исправить тоже и везде перейти на 11

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.UWP/SolutionTemplate.UWP.csproj:
```diff
@@ -98,7 +101,7 @@
     <UseVSHostingProcess>false</UseVSHostingProcess>
     <ErrorReport>prompt</ErrorReport>
     <Prefer32Bit>true</Prefer32Bit>
```
Там была тема, что переключение в `false` может помочь чтобы даже с toolchain строилось

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.UWP/SolutionTemplate.UWP.csproj:
```diff
@@ -110,7 +113,9 @@
     <ErrorReport>prompt</ErrorReport>
     <Prefer32Bit>true</Prefer32Bit>
-    <UseDotNetNativeToolchain>true</UseDotNetNativeToolchain>
+    <Use64Bitcompiler>true</Use64Bitcompiler>
```
А вот ты тут это сделал

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Wasm/GlobalSuppressions.cs:
```diff
@@ -0,0 +1,8 @@
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Wrong Usage", "DF0033:Marks undisposed objects assinged to a property, originated from a method invocation.", Justification = "<Pending>")]
```
Justification = "Disabled for WASM here because .editorconfig rule does not work"

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Wasm/LinkerConfig.xml:
```diff
@@ -0,0 +1,19 @@
+  <assembly fullname="System.Core">
+    <!-- This is required by JSon.NET and any expression.Compile caller -->
+    <type fullname="System.Linq.Expressions*" />
+  </assembly>
```
Я тебе вроде список больше присылал

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Wasm/Program.cs:
```diff
@@ -0,0 +1,25 @@
+#pragma warning disable IDE0040 // Add accessibility modifiers
+#pragma warning disable RCS1163 // Unused parameter.
+#pragma warning disable CA1801 // Review unused parameters
+        static int Main(string[] args)
```
`private` и убрать прагму

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Wasm/Program.cs:
```diff
@@ -0,0 +1,25 @@
+    public class Program
+    {
+#pragma warning disable IDE1006 // Naming Styles
+        private static App _app;
```
Переименовать. Сделай как у нас раньше было `partial App`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Wasm/SolutionTemplate.Wasm.csproj:
```diff
@@ -47,39 +44,78 @@
   </ItemGroup>
-  <Import Project="..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems" Label="Shared" />
-</Project>
+  <Import Project="..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems" Label="Shared" Condition="Exists('..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems')" />
```
Они же вроде Condition на всех импортах Shared поубирали

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.iOS/Main.cs:
```diff
@@ -0,0 +1,15 @@
+
+namespace SolutionTemplate.iOS
+{
+	public class Application
```
Аналогично `partial App`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.iOS/SolutionTemplate.iOS.csproj:
```diff
@@ -37,9 +34,9 @@
-    <ConsolePause>false</ConsolePause>
     <MtouchLink>None</MtouchLink>
     <MtouchArch>x86_64</MtouchArch>
+    <ConsolePause>false</ConsolePause>
```
Я тут настройки консолидировал по смыслу. Верни порядок

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.iOS/SolutionTemplate.iOS.csproj:
```diff
@@ -64,17 +60,13 @@
     <WarningLevel>4</WarningLevel>
-    <ConsolePause>false</ConsolePause>
-    <BuildIpa>true</BuildIpa>
-    <IOSDebugOverWiFi>true</IOSDebugOverWiFi>
```
Пусть будет

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.iOS/SolutionTemplate.iOS.csproj:
```diff
@@ -64,17 +60,13 @@
     <MtouchEnableSGenConc>true</MtouchEnableSGenConc>
-    <!-- Possible options:        -->
-    <!-- -interpreter=FSharp.Core -->
-    <!-- -linkskip=FSharp.Core    -->
```
Оставь

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.macOS/Properties/AssemblyInfo.cs:
```diff
@@ -1,21 +1,17 @@
+// General Information about an assembly is controlled through the following
 // set of attributes. Change these attribute values to modify the information
 // associated with an assembly.
 [assembly: AssemblyTitle("SolutionTemplate.macOS")]
```
Тоже с общего файла подтягивать

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.macOS/SolutionTemplate.macOS.csproj:
```diff
@@ -111,28 +118,24 @@
   <ItemGroup>
-    <Folder Include="Resources\" />
-    <Folder Include="Resources\Fonts\" />
-    <Folder Include="Assets.xcassets\unologo.imageset\" />
```
Студия почему-то эти папки выпиливает. По-моему они в шаблоне есть

dim-37 committed:
FIx all empty spaces in files

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Droid/SolutionTemplate.Droid.csproj:
```diff
@@ -0,0 +1,136 @@
+			<Compile Remove="$(_AndroidResourceDesignerFile)" />
+		</ItemGroup>
+	</Target>
+</Project>
```
Что-то мне кажется, что здесь табы

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Shared/SolutionTemplate.Shared.shproj:
```diff
@@ -1,8 +1,8 @@
-    <ProjectGuid>6279c845-92f8-4333-ab99-3d213163593c</ProjectGuid>
-    <MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>
+	<ProjectGuid>6279c845-92f8-4333-ab99-3d213163593c</ProjectGuid>
+	<MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>
```
Тут что-то не так с отступом

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Gtk/SolutionTemplate.Skia.Gtk.csproj:
```diff
@@ -0,0 +1,53 @@
+		<ProjectReference Include="..\SolutionTemplate\SolutionTemplate.fsproj" />
+	</ItemGroup>
+	<Import Project="..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems" Label="Shared" />
+</Project>
```
Отступы неправильные

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Tizen/SolutionTemplate.Skia.Tizen.csproj:
```diff
@@ -0,0 +1,57 @@
+		<ProjectReference Include="..\SolutionTemplate\SolutionTemplate.fsproj" />
+	</ItemGroup>
+	<Import Project="..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems" Label="Shared" />
+</Project>
```
Отступы неправильные

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Wpf.Host/SolutionTemplate.Skia.Wpf.Host.csproj:
```diff
@@ -0,0 +1,21 @@
+		<ProjectReference Include="..\SolutionTemplate.Skia.WPF\SolutionTemplate.Skia.WPF.csproj" />
+		<ProjectReference Include="..\SolutionTemplate\SolutionTemplate.fsproj" />
+	</ItemGroup>
+</Project>
```
Отступы неправильные

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.Skia.Wpf.Host/SolutionTemplate.Skia.Wpf.Host.csproj:
```diff
@@ -0,0 +1,21 @@
+	</PropertyGroup>
+	<ItemGroup>
+		<PackageReference Include="Uno.UI.Skia.Wpf" Version="3.7.4" />
+		<PackageReference Include="Uno.UI.RemoteControl" Version="3.7.4" Condition="'$(Configuration)'=='Debug'" />
```
Мы же тут на звёздочку?
А давай версию в props вынесем. Чтобы проще было

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate/SolutionTemplate.fsproj:
```diff
@@ -22,20 +22,26 @@
   </ItemGroup>
 
+
+
```
Много лишних строк пустых

dim-37 committed:
Fixed template in Release

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Droid/MainActivity.cs:
```diff
@@ -1,18 +1,15 @@
+    public class MainActivity : Windows.UI.Xaml.ApplicationActivity
+    {
+    }
 }
```
И по этому?

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Droid/Properties/AssemblyInfo.cs:
```diff
@@ -1,12 +1,15 @@
 // associated with an assembly.
-[assembly: AssemblyTitle(AssemblyInfo.Product + " Android app")]
-[assembly: AssemblyDescription("Android приложение" + AssemblyInfo.Description)]
+[assembly: AssemblyTitle(AssemblyInfo.Product + " Droid")]
```
Пусть будет ка было раньше

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Droid/SolutionTemplate.Droid.csproj:
```diff
@@ -2,26 +2,22 @@
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\Solution.Build.props" />
   <PropertyGroup>
-    <LangVersion>9.0</LangVersion>
```
Это мы возвращаем

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Droid/SolutionTemplate.Droid.csproj:
```diff
@@ -65,34 +63,33 @@
+		<PackageReference Include="Uno.UI.RemoteControl" Version="3.7.4" Condition="'$(Configuration)'=='Debug'" />
+		<PackageReference Include="Uno.UI" Version="3.7.4" />
+		<PackageReference Include="Uno.UniversalImageLoader" Version="1.9.33" />
+		<!--  Analyzers   -->
```
Выдели тогда их вообще в отдельную группу и поставь выше текущей группы надо строкой 65

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Droid/SolutionTemplate.Droid.csproj:
```diff
@@ -105,16 +102,20 @@
     <AndroidResource Include="Resources\values\Strings.xml" />
+  </ItemGroup>
+  <ItemGroup>
+    <AndroidResource Include="Resources\drawable\Icon.png" />
```
Консолидируй все ресурсы как я

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Shared/App.xaml.cs:
```diff
@@ -171,15 +174,17 @@ private void OnSuspending(object sender, SuspendingEventArgs e)
             var factory = LoggerFactory.Create(builder =>
             {
-//-:cnd:noEmit
+                //-:cnd:noEmit
```
А чего оно поехало?

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Shared/SolutionTemplate.Shared.shproj:
```diff
@@ -1,8 +1,8 @@
-    <ProjectGuid>6279c845-92f8-4333-ab99-3d213163593c</ProjectGuid>
-    <MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>
+	<ProjectGuid>6279c845-92f8-4333-ab99-3d213163593c</ProjectGuid>
+	<MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>
```
Тоже табы?

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Gtk/GlobalSuppressions.cs:
```diff
@@ -0,0 +1,8 @@
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Wrong Usage", "DF0033:Marks undisposed objects assinged to a property, originated from a method invocation.", Justification = "Disabled for WASM here because .editorconfig rule does not work")]
```
Так оно таки работает через supperess внутри файла проект? Если да, то нам этот файл не нужне

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Gtk/SolutionTemplate.Skia.Gtk.csproj:
```diff
@@ -1,51 +1,59 @@
+	<ItemGroup>
+		<ProjectReference Include="..\SolutionTemplate\SolutionTemplate.fsproj" />
+	</ItemGroup>
+	<Import Project="..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems" Label="Shared" />
```
Отступы неправильные. Анализаторы в отдельную группу

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Droid/SolutionTemplate.Droid.csproj:
```diff
@@ -2,26 +2,22 @@
     <FileAlignment>512</FileAlignment>
-    <AutoGenerateBindingRedirects>True</AutoGenerateBindingRedirects>
-    <UnoSourceGeneratorCaptureGenerationHostOutput>true</UnoSourceGeneratorCaptureGenerationHostOutput>
-    <UnoUIUseRoslynSourceGenerators>true</UnoUIUseRoslynSourceGenerators>
```
Верни 18 и 19. А 17 проставь через интерфейс куда оно станет там не знаю

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Tizen/SolutionTemplate.Skia.Tizen.csproj:
```diff
@@ -1,54 +1,57 @@
+	<ItemGroup>
+		<ProjectReference Include="..\SolutionTemplate\SolutionTemplate.fsproj" />
+	</ItemGroup>
+	<Import Project="..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems" Label="Shared" />
```
Аналогично

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Wpf.Host/SolutionTemplate.Skia.Wpf.Host.csproj:
```diff
@@ -1,44 +1,47 @@
+	<ItemGroup>
+		<ProjectReference Include="..\SolutionTemplate.Skia.WPF\SolutionTemplate.Skia.WPF.csproj" />
+		<ProjectReference Include="..\SolutionTemplate\SolutionTemplate.fsproj" />
+	</ItemGroup>
```
Аналогично

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Wpf/SolutionTemplate.Skia.Wpf.csproj:
```diff
@@ -1,29 +1,44 @@
+	<Import Project="..\Solution.Build.props" />
+	<PropertyGroup>
+		<TargetFramework>netstandard2.0</TargetFramework>
+		<LangVersion>9.0</LangVersion>
```
Поменяй местами 5 и 6

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Wpf/SolutionTemplate.Skia.Wpf.csproj:
```diff
@@ -1,29 +1,44 @@
+	<ItemGroup>
+		<ProjectReference Include="..\SolutionTemplate\SolutionTemplate.fsproj" />
+	</ItemGroup>
+	<Import Project="..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems" Label="Shared" />
```
Аналогично

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.UWP/Properties/Default.rd.xml:
```diff
@@ -21,11 +21,11 @@
-    
     <!-- Add your application specific runtime directives here. -->
 
 
```
Одной пустой строки достаточно

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Wasm/GlobalSuppressions.cs:
```diff
@@ -0,0 +1,8 @@
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Wrong Usage", "DF0033:Marks undisposed objects assinged to a property, originated from a method invocation.", Justification = "Disabled for WASM here because .editorconfig rule does not work")]
```
А через файл проекта не работает suppress? Если работает, то можно удалить это файл

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Wasm/Program.cs:
```diff
@@ -1,16 +1,18 @@
+﻿namespace SolutionTemplate.Wasm
 {
-    public class Program
+    public partial class Program
```
Сделай `partial App`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Wasm/SolutionTemplate.Wasm.csproj:
```diff
@@ -1,85 +1,81 @@
+		<ProjectReference Include="..\SolutionTemplate\SolutionTemplate.fsproj" />
+	</ItemGroup>
+	<Import Project="..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems" Label="Shared" Condition="Exists('..\SolutionTemplate.Shared\SolutionTemplate.Shared.projitems')" />
+</Project>
```
Аналогично

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/Info.plist:
```diff
@@ -54,5 +53,5 @@
 	<key>UIApplicationSupportsIndirectInputEvents</key>
 	<true/>
-</dict>
+  </dict>
```
Странные отступы. Надо к стати в .editorconfig добавить файлы .plist к группе XML

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/Main.cs:
```diff
@@ -1,10 +1,15 @@
+﻿using UIKit;
 
-namespace SolutionTemplate
+namespace SolutionTemplate.iOS
```
Тут прикол в том, чтобы это был partial от класса, который в `app.xaml.cs`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/Main.cs:
```diff
@@ -1,10 +1,15 @@
+namespace SolutionTemplate.iOS
 {
-    public partial class App
+    public partial class Application
```
А он называется `App`. И так должно быть во всех проектах

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/Main.cs:
```diff
@@ -1,10 +1,15 @@
     {
-        public static void Main(string[] args) =>
+        // This is the main entry point of the application.
+        static void Main(string[] args)
```
Везде нужны модификаторы доступа. У меня обычно `Main` и как `private` работал

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/Main.cs:
```diff
@@ -1,10 +1,15 @@
+        static void Main(string[] args)
+        {
+            // if you want to use a different Application Delegate class from "AppDelegate"
+            // you can specify it here.
```
Комменты не нужны, метод через лямбду

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/Properties/AssemblyInfo.cs:
```diff
@@ -1,12 +1,14 @@
 // associated with an assembly.
-[assembly: AssemblyTitle(AssemblyInfo.Product + " iOS app")]
-[assembly: AssemblyDescription("iOS приложение" + AssemblyInfo.Description)]
+[assembly: AssemblyTitle(AssemblyInfo.Product + " iOS")]
```
Сделай как было `" iOS app"`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.macOS/Main.cs:
```diff
@@ -1,15 +1,15 @@
-			NSApplication.Main(args);  
-		}
-	}
+    static class MainClass
```
Тут тоже `partial App`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.macOS/Properties/AssemblyInfo.cs:
```diff
@@ -1,21 +1,19 @@
 // associated with an assembly.
-[assembly: AssemblyTitle("SolutionTemplate.macOS")]
-[assembly: AssemblyDescription("")]
+[assembly: AssemblyTitle(AssemblyInfo.Product + " mac OS")]
```
`macOS app`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.macOS/SolutionTemplate.macOS.csproj:
```diff
@@ -1,154 +1,157 @@
+		<ItemGroup>
+			<ReferencePath Remove="@(_ReferencePathToRemove)" />
+		</ItemGroup>
+	</Target>
```
Аналогино

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.fsproj:
```diff
@@ -20,22 +20,21 @@
+		<PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="5.0.*" />
+		<PackageReference Include="Uno.Core" Version="2.3.0" />
+		<PackageReference Include="Uno.UI" Version="3.7.*" />
+		<PackageReference Include="Xamarin.Essentials" Version="1.6.*" />
```
Отступы

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate/SolutionTemplate.fsproj:
```diff
@@ -20,22 +20,21 @@
 
   <ItemGroup>
-    <PackageReference Update="FSharp.Core" Version="5.0.*" />
+    <PackageReference Update="FSharp.Core" Version="5.0.1" />
```
Возвращает на звёздочку

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Tizen/SolutionTemplate.Skia.Tizen.csproj:
```diff
@@ -2,11 +2,11 @@
     <TargetFramework>tizen50</TargetFramework>
     <OutputType>Exe</OutputType>
     <DefineConstants>$(DefineConstants);__TIZEN__;</DefineConstants>
+    <LangVersion>9.0</LangVersion>
```
Перемести на верх. Пусть у всех он будет первым пунктом внутри `PropertyGroup`

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Wpf/Program.cs:
```diff
@@ -1,4 +1,4 @@
+﻿
 namespace SolutionTemplate.Skia.Gtk
 {
 }
```
Вообще пустой `Program`?

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.Skia.Wpf/SolutionTemplate.Skia.Wpf.csproj:
```diff
@@ -2,22 +2,37 @@
   <PropertyGroup>
-    <LangVersion>9.0</LangVersion>
     <TargetFramework>netstandard2.0</TargetFramework>
+    <LangVersion>9.0</LangVersion>
```
Порядок

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.UWP/SolutionTemplate.UWP.csproj:
```diff
@@ -1,49 +1,50 @@
   <Import Project="..\Solution.Build.props" />
-  <PropertyGroup>
-    <LangVersion>9.0</LangVersion>
-    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
```
Включи через интерфейс, если это включается там

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.UWP/SolutionTemplate.UWP.csproj:
```diff
@@ -1,49 +1,50 @@
+
   <Import Project="..\Solution.Build.props" />
-  <PropertyGroup>
-    <LangVersion>9.0</LangVersion>
```
Давай это в `PropertyGroup` на 41 который начинается запишем

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/SolutionTemplate.iOS.csproj:
```diff
@@ -2,17 +2,15 @@
-    <UnoSourceGeneratorCaptureGenerationHostOutput>true</UnoSourceGeneratorCaptureGenerationHostOutput>
-    <UnoUIUseRoslynSourceGenerators>true</UnoUIUseRoslynSourceGenerators>
+    <AssemblyName>SolutionTemplate</AssemblyName>
+    <LangVersion>9.0</LangVersion>
```
Верни старый порядок

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/SolutionTemplate.iOS.csproj:
```diff
@@ -2,17 +2,15 @@
-    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
-    <UnoSourceGeneratorCaptureGenerationHostOutput>true</UnoSourceGeneratorCaptureGenerationHostOutput>
-    <UnoUIUseRoslynSourceGenerators>true</UnoUIUseRoslynSourceGenerators>
+    <AssemblyName>SolutionTemplate</AssemblyName>
```
А у нас разве в Props нет `AssemblyBaseName`?

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/SolutionTemplate.iOS.csproj:
```diff
@@ -122,50 +119,72 @@
-    <ImageAsset Visible="False" Include="Media.xcassets\AppIcons.appiconset\iPhone-20x20%403x.png" />
-    <ImageAsset Visible="False" Include="Media.xcassets\AppIcons.appiconset\iPhone-40x40%403x.png" />
-    <ImageAsset Visible="False" Include="Media.xcassets\AppIcons.appiconset\iPhone-60x60%402x.png" />
-    <ImageAsset Visible="False" Include="Media.xcassets\LaunchImages.launchimage\Contents.json" />
```
Верни как тут

xperiandri commented on src/Templates/SolutionTemplate/SolutionTemplate.iOS/SolutionTemplate.iOS.csproj:
```diff
@@ -122,50 +119,72 @@
     <Reference Include="Xamarin.iOS" />
   </ItemGroup>
   <ItemGroup>
+    <BundleResource Include="Resources\Fonts\uno-fluentui-assets.ttf" />
```
Консолидируй как я делал, строка 116 удалённая

dim-37 committed:
Fixed PR Comments

xperiandri committed:
Get rid of all the warnings and refactored csproj files

xperiandri committed:
Suppressed missing warnings and added assembly names

dim-37 committed:
Fixed Tizen manifest and macOS.csproj file

xperiandri committed:
Fixed issue on iOS and other minor issues

dim-37 committed:
Fixed NetworkStateGroup in Shell

xperiandri committed:
Moved common versions to build.props

xperiandri committed:
Fixed GitHub actions

xperiandri committed:
Fixed warnings in tests

xperiandri committed:
Implemented publish to NuGet

xperiandri committed:
Added dependency on `FSharp.Collections.Immutable`

xperiandri committed:
Added AppStore configuration for Android, set linking to SDK Only and disabled LLVM

xperiandri committed:
Added missing NoWarn for Release|iPhone

xperiandri committed:
Fixed issue with UseDotNetNativeToolchain for all platforms except x86, however crashes on runtime

xperiandri committed:
Disabled UseDotNetNativeToolchain

dim-37 committed:
Added UnoCoreVersion variable  to Solution.Build.props.

Renamed UnoVersion variable to UnoUIVersion.
Updated Uno Version from 3.7.* to 3.8.*.
Updated Uno Core Version from 2.3.0 to 2.4.*.

xperiandri committed:
Aligned output path for Android, iOS and macOS, removed root namespace from projects as it is added from props

xperiandri committed:
Fixed ICustomProperty implementation in DynamicCustomProperty

xperiandri committed:
Added sample configuration code for IOptions pattern usage

xperiandri committed:
Fixed ICustomProperty bindings initialization

xperiandri committed:
Updated NuGet packages

xperiandri committed:
Added System.Text.Json to linker

xperiandri committed:
Fixed notifications and added configuration

xperiandri committed:
Renamed Models.fs to Notification.fs and removed Notification module

xperiandri committed:
Fixed versions to work on Android, updated Uno.UI to 3.9.*

xperiandri committed:
Fixed global navigation in samples shell

xperiandri committed:
Added TextBox with two way binding and throttling

xperiandri committed:
Fixed deadlock on WASM

xperiandri committed:
Updated NuGet packages

Uno.Core to 2.4.*
Microsoft.UI.Xaml to 2.6.*
FSharp.Control.FusionTasks to 2.4.*
FSharpPlus to 1.2.*
Xamarin.Essentials to 1.7.*

xperiandri committed:
Fixed linked issue for WASM on .NET 5 and moved WASM to .NET 5

xperiandri committed:
Fixed InfoBarSeverity enum parsing issue on WASM

xperiandri committed:
Migrated core and samples projects to Solution.Build.props and aligned versions with SolutionTemplate

xperiandri committed:
Added trys with Debugger.Break() if exception for PropertyChanged and ErrorsChanged subscriptions in ViewModel

xperiandri committed:
Fixed ValidationControl bugs and improved implementation

xperiandri committed:
Fixed SubModelSeq sample containing TreeView by using WinUI version

xperiandri committed:
Implemented validation sample using ValidationControl with error styling
